sdk/storage/.editorconfig

```diff
@@ -0,0 +1,165 @@
+dotnet_style_predefined_type_for_member_access                           = false : error
+
+# Parentheses preferences
+dotnet_style_parentheses_in_arithmetic_binary_operators                  = always_for_clarity : suggestion
+dotnet_style_parentheses_in_relational_binary_operators                  = always_for_clarity : suggestion
+dotnet_style_parentheses_in_other_binary_operators                       = always_for_clarity : suggestion
+dotnet_style_parentheses_in_other_operators                              = never_if_unnecessary : suggestion
+
+# Modifier preferences
+dotnet_style_require_accessibility_modifiers                             = never : suggestion
+dotnet_style_readonly_field                                              = true : error
+
+# Expression-level preferences
+dotnet_style_object_initializer                                          = true : suggestion
+dotnet_style_collection_initializer                                      = true : suggestion
+dotnet_style_explicit_tuple_names                                        = true : suggestion
+dotnet_style_null_propagation                                            = true : suggestion
+dotnet_style_coalesce_expression                                         = true : suggestion
+dotnet_style_prefer_is_null_check_over_reference_equality_method         = true : suggestion
+dotnet_style_prefer_inferred_tuple_names                                 = true : error
+dotnet_style_prefer_inferred_anonymous_type_member_names                 = true : suggestion
+dotnet_style_prefer_auto_properties                                      = true : suggestion
+dotnet_style_prefer_conditional_expression_over_assignment               = true : suggestion
+dotnet_style_prefer_conditional_expression_over_return                   = true : suggestion
+
+###############################
+# Naming Conventions          #
+###############################
+
+# Style Definitions
+dotnet_naming_style.pascal_case_style.capitalization                     = pascal_case
+
+# Use PascalCase for constant fields
+dotnet_naming_rule.constant_fields_should_be_pascal_case.severity        = suggestion
+dotnet_naming_rule.constant_fields_should_be_pascal_case.symbols         = constant_fields
+dotnet_naming_rule.constant_fields_should_be_pascal_case.style           = pascal_case_style
+dotnet_naming_symbols.constant_fields.applicable_kinds                   = field
+dotnet_naming_symbols.constant_fields.applicable_accessibilities         = *
+dotnet_naming_symbols.constant_fields.required_modifiers                 = const
+
+###############################
+# C# Coding Conventions       #
+###############################
+
+# var preferences
+csharp_style_var_for_built_in_types                                      = true : warning
+csharp_style_var_when_type_is_apparent                                   = true : warning
+csharp_style_var_elsewhere                                               = true : warning
+

```

Comment:
I'd like to enforce these.

Reply:
specifically because we don't want it to bifurcate -- picking a single one, I'd pick var.

Reply:
These are copied as-is from the old repo, so there's no change from what it used to be.  If we want to strengthen, etc., I'd prefer to do that as a separate change because this PR is already incredibly noisy.